packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -463,16 +464,15 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<TDate>(
     [calendars],
   );
 
-  const visibleMonths = React.useMemo(
-    () =>
-      Array.from({ length: calendars }).map((_, index) =>
-        utils.setMonth(
-          calendarState.currentMonth,
-          utils.getMonth(calendarState.currentMonth) + index,
-        ),
-      ),
-    [utils, calendarState.currentMonth, calendars],
-  );
+  const visibleMonths = React.useMemo(() => {
+    const cleanCurrentMonthCalendarPosition = Math.min(currentMonthCalendarPosition, calendars);
```

Comment:
I would have added a warning in the console if `calendars` and `currentMonthCalendarPosition` are incoherent. But works too 👍

Reply:
Done :+1: